### DISCUSSION TITLE
Docs for dropdown is id required or just missing

### DISCUSSION THREAD

**Discussion by thewebartisan7** [twbs/bootstrap#35755] [General]
In the docs the id for dropdown it's used but not in example for dropup, dropend and dropstart, see:
https://getbootstrap.com/docs/5.1/components/dropdowns/#dropright

Does this mean the id it's not required and just for accessibility for aria-labelledby ?

**Comment by mdo** [comment]
@patrickhlauke We should probably update these to include `id` and aria attributes, no?

**Reply by patrickhlauke** [reply]
yeah, for consistency we should add `id` and `aria-labelledby` to those other drop examples. i'll get onto that.

they're not required to make the dropdown etc work correctly, if that was the question @thewebartisan7 ... and even for accessibility, they're more a "nice to have" than a hard requirement.

**Reply by patrickhlauke** [reply]
then again, wonder if we by default not use `id` / `aria-labelledby`, but give an example clearly marked as "for accessibility, you may consider doing this". it's not necessary, more of a question of taste...

**Reply by mdo** [reply]
I'm down for that!

**Reply by jrchamp** [reply]
> they're not required to make the dropdown etc work correctly, if that was the question @thewebartisan7 ... and even for accessibility, they're more a "nice to have" than a hard requirement.

If it's optional, but "nice to have" that means it's _better_ to have it, right? So removing it in #36487 is therefore somewhat _worse_? As accessibility is a legal requirement for many of us, it would be nice if the examples reflected the accessible version. Maybe by "question of taste" you mean that it doesn't necessarily improve or reduce accessibility either way, however, I don't see anywhere in the changes an explanation (such as an example) that provides guidance. Based on the code changes, it seems like the recommendation is to remove these two attributes which are "confusing", though it's unclear to me if it's confusing for users of AT (which would be very bad, and it makes sense to fix the problem) or if it's confusing to developers (like me, who might end up at this discussion looking for an explanation).

I hope this helps clarify what this all looks like to users like me who are not in the loop. Thank you all for your amazing efforts to build the tools that make everything better.
